apps/core/src/utils/stake/parseEventJson.ts

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import type { IotaEvent } from '@iota/iota-sdk/client';
+import { ParsedJson } from '../../interfaces';
+
+export function parseEventJson<T extends ParsedJson>(event: IotaEvent): T {
```

Comment:
you are not really parsing but rather getting the parsed json, I would probably rename this function to `getParsedJson` or even probably remove it and make getStakeDetails & getUnstakeDetails be based on the event and not the parsed json, to avoid 1 extra conversion that can happen directly in a single function eg `getStakeDetailsFromEvent(event)`